src/arch/arm/tracers/tarmac_tracer.cc

```diff
@@ -64,9 +64,9 @@ tarmacDump(const TarmacTracerParams &p)
 {
     switch (p.outfile) {
       case TarmacDump::stdoutput:
-        return simout.findOrCreate("stdout");
+        return simout.findOrCreate("stdout.txt");
       case TarmacDump::stderror:
-        return simout.findOrCreate("stderr");
+        return simout.findOrCreate("stderr.txt");
```

Comment:
I believe this is not working as you would imagine. The findOrCreate argument is matched against the stdout and stderr literals to check where to redirect the output stream:

```
OutputStream *
OutputDirectory::checkForStdio(const std::string &name)
{
    if (name == "cerr" || name == "stderr")
        return &stderr;

    if (name == "cout" || name == "stdout")
        return &stdout;

    return NULL;
}
```

https://github.com/gem5/gem5/blob/stable/src/base/output.cc#L135

By passing stdout.txt/stderr.txt comparison will fail and gem5 will always redirect to the txt files. I believe this change here should be reverted. The redirection to the txt file will be handled automatically by the rest of the code

Reply:
Thanks @giactra! I don't think I would've ever figured that out by myself.

This is fixed.